### ISSUE TITLE
"Attempting to use a disconnected port object"

### ISSUE THREAD

**Issue by raszpl** [Tampermonkey/tampermonkey#2107]
flood of 
```
Error: Attempting to use a disconnected port object
    at content.js:46:437
    at content.js:16:256
    at Array.forEach (<anonymous>)
    at Mt (content.js:8:1214)
    at t (content.js:9:295)
    at Object.message (content.js:16:247)
    at Object.message (content.js:17:269)
    at content.js:75:202
    at _ (content.js:68:376)
    at Mt (content.js:8:1214)
```
in console. I think I also saw "port disconnected something something one pointing as similar place in content.js
Noticed while debugging another Extension as I was spending a ton of time in devtools. Doesnt happen always, and doesnt happen immediately.
Its entirely possible to just be a Vivaldi bug, as I also noticed another very weird bug while heavily restarting another extension using messages - at some point in a 48h browser session browser Extension Toolbar started growing with duplicated Tempermonkey "action" icons on every restart of that another extension (Improvetube).

EDIT: I seem to remember  reporting something similar and it turned out to be linked to extension being updated invalidating old background.js? so I checked and latest update was 5.3.6202 2024-06-24 yesterday. Possible its just this combined with browser running for couple of days non stop.

## Specifications
Vivaldi | 6.8.3381.3 (Chrome/126.0.0.0 Safari/537.36)
- TM: v5.3.6201
- OS: win10

**Comment by raszpl** [comment]
Still seeing this whenever I debug some background.js scripts and pause JS execution. Is it possible pausing background script on one Extension is also pausing all the other background scripts including Tampermonkey? Looking at content.js:46:437 its right around keepalive so maybe when I debug something and pause I make tampermonkey keepalive timeout?

**Comment by derjanb** [comment]
I was unable to reproduce ths issue, but I hope to have it fixed at 5.3.6204 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx))

Please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions` (after you've enabled 'Developer Mode').

**Comment by raszpl** [comment]
Im already using gcalenpjmijncebpfijmoaglllgpjagf and auto updated to BETA 5.3.6204 a week ago?
I was able to find a way to reproduce. Nothing to do with background, just pause javascript execution in main website thread for couple minutes, no js runs = TM injected script doesnt work = doesnt respond to keepalive. In the end its a non issue, will only trigger for someone sitting in devtools debugger messing with his own extension. Sorry for wasting your time :(
